public/examples/card-on-file-payment-with-verification.html

```diff
@@ -0,0 +1,183 @@
+        const bodyParameters = {
+          locationId,
+          sourceId,
+          customerId,
+        };
+
+        if (verificationToken !== undefined) {
+          bodyParameters.verificationToken = verificationToken;
+        }
+
+        const body = JSON.stringify(bodyParameters);
+
+        const paymentResponse = await fetch('/payment', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body,
+        });
+
+        if (paymentResponse.ok) {
+          return paymentResponse.json();
+        }
+
+        const errorBody = await paymentResponse.text();
+        throw new Error(errorBody);
+      }
+
+      async function tokenize(paymentMethod) {
+        const tokenResult = await paymentMethod.tokenize();
+        if (tokenResult.status === 'OK') {
+          return tokenResult.token;
+        } else {
+          throw new Error(
+            `Tokenization errors: ${JSON.stringify(tokenResult.errors)}`
+          );
+        }
+      }
+
+      // status is either SUCCESS or FAILURE;
+      function displayPaymentResults(status) {
+        const statusContainer = document.getElementById(
+          'payment-status'
+        );
+        if (status === 'SUCCESS') {
+          statusContainer.classList.remove('is-failure');
+          statusContainer.classList.add('is-success');
+        } else {
+          statusContainer.classList.remove('is-success');
+          statusContainer.classList.add('is-failure');
+        }
+
+        statusContainer.style.visibility = 'visible';
+      }
+
+      async function verifyBuyer(payments, sourceId) {
+        const verificationDetails = {
+          amount: '1.00',
+          billingContact: {
+            addressLines: ['123 Main Street', 'Apartment 1'],
+            familyName: 'Doe',
+            givenName: 'John',
+            email: 'jondoe@gmail.com',
+            country: 'GB',
+            phone: '3214563987',
+            region: 'LND',
+            city: 'London',
+          },
+          currencyCode: 'GBP',
+          intent: 'CHARGE',
+        };
+
+        const verificationResults = await payments.verifyBuyer(
+          sourceId,
+          verificationDetails
+        );
+        return verificationResults.token;
+      }
+
+      document.addEventListener('DOMContentLoaded', async function () {
+        if (!window.Square) {
+          throw new Error('Square.js failed to load properly');
+        }
+
+        let payments;
+        try {
+          payments = window.Square.payments(appId, locationId);
+        } catch {
+          const statusContainer = document.getElementById(
+            'payment-status'
+          );
+          statusContainer.classList.add('missing-credentials');
+          statusContainer.style.visibility = 'visible';
+          return;
+        }
+
+        async function handlePaymentMethodSubmission(
```

Comment:
How about we update the parameter list to only include: event, shouldVerify. So we exclude paymentMethod since it's no longer used. Then we can extract the input values within this method to pass to createPayment.

Reply:
We can also just make a different function to handle this, as the flow is very different from the other payment methods.